### PR TITLE
Feature/invocable naming

### LINKED ISSUE #1837: Invocable method for Opportunity Naming
[Per this hub thread](https://powerofus.force.com/0D58000002NuYqp) it would be nice to be able to rename an opportunity programatically.

### PR DESCRIPTION
**Pull request by cdcarter**
This is closing out the last two PRs for a combined "invocable naming" feature.

These global classes use the _Action suffix instead of _API to stay consistent with SFDC docs and examples regarding Invocable Methods. 

Though labels DIDN'T get weird this time around, some how my setup clobbered the whitespace around HH_HouseholdNaming_TEST. Not sure what that is about, but at least it's mergeable.

The test class is getting repetitive, and if you don't want to merge the tests as is, I would understand that. The class is just a minor wrapper around the tested code, so the test method provided is purely for coverage reasons. As more global interfaces to NPSP functionality are created, I'd love some guidance on what replication in tests is a-ok/what should be split out a little bit/where real love is needed.
# Contribution Info

This external contribution was submitted by @cdcarter and Fixes #1837 and #864.
# Changes
- Adds a global @InvocableMethod class OPP_OpportunityNaming_Action to provide a method to force opportunity renaming programmatically (via Apex, Flow, or PB).
- Adds a global @InvocableMethod class HH_HouseholdNaming_Action to provide a method to force opportunity renaming programmatically (via Apex, Flow, or PB).

### CODE REVIEW

**Comment by davidhabib** [comment]
@mpusto The story points for integrating this is probably 5.  The bigger question is the team spending the time to plan out an invocable action strategy.  What do we want to expose?  How should it look?  Are these methods the ones we want?  I'd suggest we add a user story to do that strategy work before considering this pull request.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 1]
```diff
@@ -41,13 +41,13 @@ public class HH_HouseholdNaming_TEST {
     * if you want to run all tests, then use '*'
     */
     private static string strTestOnly = '*';
-    
+
     /*********************************************************************************************************
     * @description Test method for batch
     */
     public static testMethod void testBatch() {
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: I suggest to annotate test methods with `@IsTest` for better readability.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: I suggest to annotate test methods with `@IsTest` for better readability.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 2]
```diff
@@ -167,42 +223,42 @@ public class HH_HouseholdNaming_TEST {
...
                 npe01__Opportunity_Contact_Role_Default_role__c = 'Donor'
             ));
-        
+
         integer i;
-        
+
         list<Contact> insertlist = new list<Contact>();
         for (i = 0; i < 20; i++){
             Contact c = new Contact(LastName = 'L' + i, FirstName = 'F' + i, Salutation = 'Mr.');
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 3]
```diff
@@ -109,50 +109,106 @@ public class HH_HouseholdNaming_TEST {
...
+                    npo02__Excluded_Contact_Opp_Types__c = null
+                ));
+
+            npe01__Contacts_and_Orgs_Settings__c contactSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_and_Orgs_Settings__c (
+                npe01__Account_Processor__c = CAO_Constants.ONE_TO_ONE_PROCESSOR,
+                npe01__Enable_Opportunity_Contact_Role_Trigger__c = true,
+                npe01__Opportunity_Contact_Role_Default_role__c = 'Donor'
+            ));
+
+        integer i;
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 4]
```diff
@@ -109,50 +109,106 @@ public class HH_HouseholdNaming_TEST {
...
+
+        list<Contact> insertlist = new list<Contact>();
+        for (i = 0; i < 20; i++){
+            Contact c = new Contact(LastName = 'LastName' + i, FirstName = 'FirstName' + i, Salutation = 'Mr.');
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 5]
```diff
@@ -109,50 +109,106 @@ public class HH_HouseholdNaming_TEST {
...
-                
+
             npe01__Contacts_and_Orgs_Settings__c contactSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_and_Orgs_Settings__c (
                 npe01__Account_Processor__c = CAO_Constants.ONE_TO_ONE_PROCESSOR,
                 npe01__Enable_Opportunity_Contact_Role_Trigger__c = true,
                 npe01__Opportunity_Contact_Role_Default_role__c = 'Donor'
             ));
-        
+
         integer i;
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 6]
```diff
@@ -109,50 +109,106 @@ public class HH_HouseholdNaming_TEST {
...
-        system.assertEquals('LastName0 ' +  system.label.npo02.DefaultHouseholdName, con.npo02__household__r.Name);  
+        system.assertEquals('LastName0 ' +  system.label.npo02.DefaultHouseholdName, con.npo02__household__r.Name);
         system.assertEquals('FirstName0',con.npo02__household__r.npo02__Informal_Greeting__c);
         system.assertEquals('Mr. FirstName0 LastName0', con.npo02__household__r.npo02__Formal_Greeting__c);
-        
+
     }
-    
+
+    public static testMethod void HouseholdNamingActionTest(){
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: * I suggest to annotate test methods with `@IsTest` for better readability.
* Methods should use the `lowerCamelCase` naming convention. Please rename accordingly.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: * Methods should use the `lowerCamelCase` naming convention. Please rename accordingly.
* I suggest to annotate test methods with `@IsTest` for better readability.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 7]
```diff
@@ -167,42 +223,42 @@ public class HH_HouseholdNaming_TEST {
...
-                
+
             npe01__Contacts_and_Orgs_Settings__c contactSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_and_Orgs_Settings__c (
                 npe01__Account_Processor__c = CAO_Constants.ONE_TO_ONE_PROCESSOR,
                 npe01__Enable_Opportunity_Contact_Role_Trigger__c = true,
                 npe01__Opportunity_Contact_Role_Default_role__c = 'Donor'
             ));
-        
+
         integer i;
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

File: src/classes/HH_HouseholdNaming_TEST.cls [hunk 8]
```diff
@@ -109,50 +109,106 @@ public class HH_HouseholdNaming_TEST {
...
-        
+
         list<Contact> insertlist = new list<Contact>();
         for (i = 0; i < 20; i++){
             Contact c = new Contact(LastName = 'LastName' + i, FirstName = 'FirstName' + i, Salutation = 'Mr.');
```
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

File: src/classes/HH_HouseholdNaming_TEST.cls (hunk 7 above)
Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

Comment (LurchTheButler) [inline src/classes/HH_HouseholdNaming_TEST.cls]: This variable doesn't seem to have a very insightful name. Please give variables names that suggest their purpose.

**Comment by davidhabib** [comment]
@mpusto @cdcarter we've had this pull request open for almost 15 months now!  As originally stated, I'm not comfortable adding these two invocable methods until we come up with a formal strategy on what things we want to expose thru API's, REST, invocable methods, etc.

@mpusto do you know if we have such a backlog item?

@cdcarter are you able to preserve your work, if we close this, so that when we are ready to work on this you can resurrect the code and/or pull request?

@dylanhansen fyi

**Comment by cdcarter** [comment]
Several of the usecases that originally prompted the need for these as invocables have been resolved in other ways.

I am closing this. It's waaay off master, the tests are icky to say the least, and i can stash the branch on my fork.

**Pull request closed without merging**